clients/src/main/java/org/apache/kafka/clients/consumer/ConsumerRecord.java

```diff
@@ -296,13 +223,23 @@ public Optional<Integer> leaderEpoch() {
         return leaderEpoch;
     }
 
+    /**
+     * Get the delivery count for the record if available
+     *
+     * @return the delivery count or empty when deliveries not counted
```

Comment:
out of curiosity: Is `Optional.of(0)` equal to `Optional.empty`? Both of them means "deliveries not counted"

Reply:
I would say they are not the same. `Optional.empty()` means deliveries are not being counted. `Optional.of(0)` would mean they are being counted and this is the zeroth delivery.

If we were using Short instead, it would be the difference between `null` and `Short(0)`, that is, no value and a value of zero.

Reply:
Sure. I will do that.